### PR TITLE
chore: scaffold documentation

### PR DESCRIPTION
**Pull request by iherger**
### Detailed summary
- Added documentation for the Enzyme API and SDK in the `docs` folder.
- Updated `pnpm-workspace.yaml` to include the `docs` folder.
- Updated `tsconfig.json` to include the `docs` folder.
- Added scripts and dependencies for the `docs` package in `docs/package.json`.
- Added TypeScript configuration for the `docs` folder in `docs/tsconfig.json`.
- Added new MDX files for various API endpoints and SDK actions in the `docs/pages` folder.
- Updated the `vocs.config.ts` file to include new routes for the documentation.
- Added new code examples for depositing and withdrawing from an Enzyme vault in the `docs/pages/sdk/depositor` folder.
- Added new code examples for interacting with the Enzyme API in the `docs/pages/api` folder.

> The following files were skipped due to too many changes: `docs/pages/sdk/depositor/deposit.mdx`, `docs/pages/sdk/overview.mdx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@enzymefinance/api-sdk-documentation",
+  "version": "0.0.0",
+  "type": "module",
```
Comment (fubhy) [inline docs/package.json:4]: this has to be marked as `private: true`. Otherwise `changesets` will publish it on `npm`.

Comment (fubhy) [reply docs/package.json:4]: We also have to add `@enzymefinance/docs` to the `ignore` list in `.changesets/config`.

File: docs/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@enzymefinance/api-sdk-documentation",
```
Comment (fubhy) [inline docs/package.json:2]: Rename this to `@enzymefinance/docs`

**Review by fubhy — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "test": "vitest",
     "test:coverage": "vitest --coverage",
     "typecheck": "tsc --build tsconfig.json",
+    "documentation": "pnpm run --filter @enzymefinance/api-sdk-documentation dev",
```
Comment (fubhy) [inline package.json:12]:
```suggestion
    "documentation": "pnpm run --filter @enzymefinance/docs dev",
```

File: docs/package.json (hunk 2 above)
Comment (fubhy) [inline docs/package.json:2]:
```suggestion
  "name": "@enzymefinance/docs",
```

File: docs/package.json (hunk 1 above)
Comment (fubhy) [inline docs/package.json:4]:
```suggestion
  "private": true,
  "type": "module",
```

**Review by fubhy — CHANGES_REQUESTED** [review]

File: docs/package.json (hunk 1 above)
Comment (iherger) [reply docs/package.json]: Done.

**Review by fubhy — CHANGES_REQUESTED** [review]

**Pull request merged**